rust/src/measurements/permute_and_flip/mod.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    core::{Function, Measurement, PrivacyMap},
+    domains::{AtomDomain, VectorDomain},
+    error::Fallible,
+    measurements::Optimize,
+    measures::MaxDivergence,
+    metrics::LInfDistance,
+    traits::samplers::{sample_bernoulli_exp, CastInternalRational, SampleUniformIntBelow},
+    traits::{DistanceConstant, Float, Number},
+};
+use dashu::base::Sign;
+use dashu::rational::RBig;
+
+fn exact_fisher_yates(n: usize, trials_per_coin: Option<usize>) -> Fallible<Vec<usize>> {
```

Comment:
* I think it would be better to write this as a function that shuffles a slice
* I don't think you need to plumb the number of trials up, unless you have a good sense for how you might handle the error and still keep this algorithm timing-safe.

I think this signature then becomes:
`fn exact_fisher_yates<T>(&mut [T] slice) -> Fallible<()>`

Calling like:
```
let mut permutation = (0..arg.len()).collect();
exact_fisher_yates(permutation.as_mut_slice())?;
```